### ISSUE TITLE
Difference between bart-large and bart-large-cnn vocabulary

### ISSUE THREAD

**Issue by swethmandava** [huggingface/transformers#7346]
Trying to finetune from pretrained bart checkpoint as follows:

```
config = BartConfig(**json.load(open(args.config_path, "r"))) #pointing to bart-large-cnn/config.json
model = BartForConditionalGeneration.from_pretrained('facebook/bart-large', config=config) #use pretrained bart model's weights
tokenizer = BartTokenizer.from_pretrained('facebook/bart-large-cnn')
```

but since facebook/bart-large and facebook/bart-large-cnn have different vocab size, it fails. What's the reason behind different vocab sizes? How can I use pretrained bart for finetuning - should I modify bart-large-cnn's config to use the same vocab size as bart-large?

@sshleifer

**Comment by sshleifer** [comment]
The reason is the mask token, see https://github.com/huggingface/transformers/issues/3108.
You could try to use the resize_token_embeddings method, but even easier would be to pass the config changes you want to init
```python
BartForConditionalGeneration.from_pretrained('bart-large', num_beams=4, min_length=56, max_length=142, length_penalty=142, ...)
```
